### ISSUE TITLE
Docs/tutorials/samples ?

### ISSUE THREAD

**Issue by rishavs** [liquid600pgm/rapid#14]
Hi

Are you planning to add

- ref docs
- general tutorials
- sample demos

for the engine?

**Comment by liquid600pgm** [comment]
Sure, reference documentation is already written in the source files, so it should only be a matter of generating it with `nim doc`, but examples and demos would still have to be written. 
The readme includes a few very basic demos, but they're not sufficient for understanding how to make a game with rapid. I'll try to add some more docs when I find some time.

**Comment by rishavs** [comment]
Thanks @liquid600pgm 

a demo on structuring a basic game will be most helpful as thats where I struggle the most as a novice iuser. Does Rapid comes with a scene manager?

**Comment by liquid600pgm** [comment]
rapid doesn't come with a scene manager, it's mostly made for my way of coding, so the "idiomatic" way would be a State enum or multiple RWorlds.

Also, please leave this open, I easily forget things.

**Comment by rishavs** [comment]
Would it better if I add separate issue for each item with more details?

**Comment by liquid600pgm** [comment]
No need to do so, it's fine the way it is.

**Comment by rishavs** [comment]
checking back in after an year. ^_^

Would really love to try out rapid but without some starting docs/tuts, it is hard for me to do so. Hope you will be able to put something together for interested folks. this engine looks way too interesting to leave alone. 😄

**Comment by liquidev** [comment]
@rishavs I'll try to cobble an introductory tutorial or something once the new version is finished. I'm busy with school and other projects now though, and I haven't worked on rapid for a while now.

**Comment by rishavs** [comment]
hah no worries. I know life comes in the way! 😃

**Comment by landgazr** [comment]
Hello! Just tried compiling the tests as-is with `Nim Compiler Version 1.6.2 [Linux: amd64]`. All work except for `tchipmunk`, but that seems to be known per the commit messages, and is a `glm` problem anyway...?

That said, since there are code-based refdocs and (mostly) working tests, all that remains is an intro tutorial right? I'm working on a basic game using this package and I could parlay my experience into one.

**Comment by liquidev** [comment]
Hey, I'm not sure what's preventing tchipmunk from compiling. It would be nice if you could file a separate issue about that, including the error message.

About the intro tutorial, feel free to write one if you want to; I'm willing to accept PRs for documentation and such.
